File: screenshot.js
// screenshot.js
#!/usr/bin/env node
'use strict';

const screenshot = require('screenshot-desktop');
const fs = require('fs');
const path = require('path');

function parseArgs() {
    const args = process.argv.slice(2);
    const opts = { output: null, delay: 0, region: null };
    for (let i = 0; i < args.length; i++) {
        if (args[i] === '-o' || args[i] === '--output') {
            opts.output = args[++i];
        } else if (args[i] === '-d' || args[i] === '--delay') {
            opts.delay = parseInt(args[++i], 10);
        } else if (args[i] === '-r' || args[i] === '--region') {
            opts.region = args[++i];
        } else if (args[i] === '-h' || args[i] === '--help') {
            console.log(`Usage: node screenshot.js [options]
Options:
  -o, --output <file>  Выходной файл
  -d, --delay <sec>    Задержка в секундах
  -r, --region <x,y,w,h> Область захвата
  -h, --help           Справка`);
            process.exit(0);
        }
    }
    return opts;
}

async function main() {
    const opts = parseArgs();
    if (opts.delay > 0) {
        console.log(`Ожидание ${opts.delay} секунд...`);
        await new Promise(resolve => setTimeout(resolve, opts.delay * 1000));
    }
    let output = opts.output;
    if (!output) {
        const now = new Date();
        const ts = now.toISOString().replace(/[-:.]/g, '').slice(0, 14);
        output = `screenshot_${ts}.png`;
    }
    let options = {};
    if (opts.region) {
        const parts = opts.region.split(',').map(Number);
        if (parts.length !== 4) {
            console.error('Неверный формат области. Используйте x,y,w,h');
            process.exit(1);
        }
        // screenshot-desktop не поддерживает область напрямую, используем screen с последующим обрезанием.
        // Для простоты будем захватывать весь экран и обрезать с помощью sharp, но это дополнительная зависимость.
        // Вместо этого выведем предупреждение.
        console.warn('Регион не поддерживается в этой версии, захват всего экрана.');
    }
    try {
        const imgBuffer = await screenshot({ format: 'png' });
        fs.writeFileSync(output, imgBuffer);
        console.log(`Скриншот сохранён в ${output}`);
    } catch (err) {
        console.error(`Ошибка: ${err.message}`);
        process.exit(1);
    }
}

main();

File: screenshot.java
// screenshot.java
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;
import javax.imageio.ImageIO;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.TimeUnit;

public class screenshot {
    public static void main(String[] args) {
        String output = null;
        int delay = 0;
        String region = null;
        for (int i = 0; i < args.length; i++) {
            if (args[i].equals("-o") && i+1 < args.length) {
                output = args[++i];
            } else if (args[i].equals("-d") && i+1 < args.length) {
                delay = Integer.parseInt(args[++i]);
            } else if (args[i].equals("-r") && i+1 < args.length) {
                region = args[++i];
            } else if (args[i].equals("-h") || args[i].equals("--help")) {
                System.out.println("Usage: java screenshot [options]");
                System.out.println("  -o <file>   Output file");
                System.out.println("  -d <sec>    Delay in seconds");
                System.out.println("  -r <x,y,w,h> Region");
                System.out.println("  -h          Help");
                return;
            }
        }
        if (delay > 0) {
            System.out.println("Ожидание " + delay + " секунд...");
            try {
                TimeUnit.SECONDS.sleep(delay);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        if (output == null) {
            String ts = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
            output = "screenshot_" + ts + ".png";
        }

        try {
            Robot robot = new Robot();
            BufferedImage image;
            if (region != null) {
                String[] parts = region.split(",");
                if (parts.length != 4) {
                    System.err.println("Неверный формат области. Используйте x,y,w,h");
                    System.exit(1);
                }
                int x = Integer.parseInt(parts[0]);
                int y = Integer.parseInt(parts[1]);
                int w = Integer.parseInt(parts[2]);
                int h = Integer.parseInt(parts[3]);
                Rectangle rect = new Rectangle(x, y, w, h);
                image = robot.createScreenCapture(rect);
            } else {
                Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
                Rectangle screenRect = new Rectangle(screenSize);
                image = robot.createScreenCapture(screenRect);
            }
            ImageIO.write(image, "png", new File(output));
            System.out.println("Скриншот сохранён в " + output);
        } catch (Exception e) {
            System.err.println("Ошибка: " + e.getMessage());
            System.exit(1);
        }
    }
}
